frontend/api/createAxiosInstance.ts

```diff
@@ -1,13 +1,30 @@
-import axios from "axios";
+import axios, { AxiosInstance } from "axios";
 
-const createAxiosInstance = () => {
+const createAxiosInstance = (): AxiosInstance => {
 	const config = {
 		baseURL: window.location.origin,
 		timeout: 30000,
 		withCredentials: true,
 	};
 
-	return axios.create(config);
+	const instance = axios.create(config);
+
+	instance.interceptors.response.use(
+		(response) => response,
+		(error) => {
+			const STATUS_UNAUTHORIZED = 401;
+			if (error?.response?.status === STATUS_UNAUTHORIZED) {
+				// Server's 302 redirection only works on documents, and not xhr calls.
+				const path =
+					error?.response?.data?.redirectPath ?? "/shelter/login";
+				console.log(`Unauthenticated API call. Redirecting to ${path}`);
+				window.location.assign(path);
+			}
+			return error;
```

Comment:
returning here seems to swallow errors; the promise resolves with the error and the caller will continue as though the request was successful

Reply:
I see. no wonder the axios example used Promise.reject. changing it.